packages/flutter/lib/src/material/user_accounts_drawer_header.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:flutter/widgets.dart';
+
+import 'debug.dart';
+
+/// The top-most region of a material design drawer with user accounts. The
+/// header's [decoration] is used to provide a background.
+/// [currentAccountPicture] is the main account picture on the left, while
+/// [otherAccountsPictures] are the smaller account pictures on the right.
+/// [accountName] and [accountEmail] provide access tothe top and bottom rows
+/// of the account details in the lower part of the header. When touched, this
+/// area triggers [onDetailsPressed] and toggles the dropdown icon on the right.
+///
+/// Part of the material design [Drawer].
+///
+/// Requires one of its ancestors to be a [Material] widget.
+///
+/// See also:
+///
+///  * [Drawer]
+///  * [DrawerItem]
+///  * <https://www.google.com/design/spec/patterns/navigation-drawer.html>
+
+class UserAccountsDrawerHeader extends StatefulWidget {
+  /// Creates a material design drawer header.
+  ///
+  /// Requires one of its ancestors to be a [Material] widget.
+  UserAccountsDrawerHeader({
+    Key key,
+    this.decoration,
+    this.currentAccountPicture,
+    this.otherAccountsPictures,
+    this.accountName,
+    this.accountEmail,
+    this.onDetailsPressed
+  }) : super(key: key);
+
+  /// A callback that gets called when the account name/email/dropdown
+  /// section is pressed.
+  final VoidCallback onDetailsPressed;
+
+  /// Decoration for the main drawer header container useful for applying
+  /// backgrounds.
+  final BoxDecoration decoration;
+
+  /// A widget placed in the upper-left corner representing the current
+  /// account picture. Normally a [CircleAvatar].
+  final Widget currentAccountPicture;
+
+  /// A widget list placed in the upper-right corner representing
+  /// pictures of other accounts. Normally a list of [CircleAvatar] widgets.
+  ///
+  /// Up to 3 pictures displayed.
+  final List<Widget> otherAccountsPictures;
+
+  /// A widget placed on the top row of the account details representing
+  /// account name.
+  final Widget accountName;
+
+  /// A widget placed on the bottom row of the account details representing
+  /// account email.
+  final Widget accountEmail;
+
+  @override
+  _UserAccountsDrawerHeaderState createState() =>
+      new _UserAccountsDrawerHeaderState();
+}
+
+class _UserAccountsDrawerHeaderState extends State<UserAccountsDrawerHeader> {
+  /// Saves whether the account dropdown is open or not.
+  bool isOpen = false;
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMaterial(context));
+    return new DrawerHeader(
+      decoration: config.decoration,
+      child: new Column(
+        children: <Widget>[
+          new Flexible(
+            child: new Stack(
+              children: <Widget>[
+                new Positioned(
+                  top: 0.0,
+                  right: 0.0,
+                  child: new Row(
+                    children: config.otherAccountsPictures.take(3).map(
```

Comment:
otherAccountsPictures may be null so `config.otherAccountsPictures?.take(3).map`